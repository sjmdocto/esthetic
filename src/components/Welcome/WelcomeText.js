import React from 'react';
import {Text, View} from 'react-native';
import {styles} from './WelcomeText.style';

/**
 * @function WelcomeText
 * @param {*} props
 * @param {boolean} props.visible
 */
const WelcomeText = (props) => {
  if (!props.visible) {
    return null;
  }
  let line1 = 'Welcome to Esthetic, your new virtual closet.\n';
  let line2 =
    'To get started, just click on the plus in the top right to add your clothes!';
  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        {line1}
        {line2}
      </Text>
    </View>
  );
};

export default WelcomeText;
