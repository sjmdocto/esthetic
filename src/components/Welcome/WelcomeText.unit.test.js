import React from 'react';
import WelcomeText from './WelcomeText';
import {render} from 'react-native-testing-library';

describe('WelcomeText test suite:', () => {
  test('If visible prop is equal to false, WelcomeText is NOT rendered.', () => {
    const isVisible = false;
    const {toJSON} = render(<WelcomeText visible={isVisible} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('If visible prop is equal to true, WelcomeText is rendered.', () => {
    const isVisible = true;
    const {toJSON} = render(<WelcomeText visible={isVisible} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
