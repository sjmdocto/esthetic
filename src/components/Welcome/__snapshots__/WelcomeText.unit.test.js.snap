// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeText test suite: If visible prop is equal to false, WelcomeText is NOT rendered. 1`] = `null`;

exports[`WelcomeText test suite: If visible prop is equal to true, WelcomeText is rendered. 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E5E5E5",
      "flex": 0,
      "height": 667,
      "justifyContent": "center",
      "paddingHorizontal": 75,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "fontSize": 14,
        "fontStyle": "italic",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    Welcome to Esthetic, your new virtual closet.

    To get started, just click on the plus in the top right to add your clothes!
  </Text>
</View>
`;
